### PR TITLE
adding with-axiom example

### PR DESCRIPTION
**Pull request by islameldigwi**
## Documentation / Examples

- [ X ] Make sure the linting passes by running `pnpm lint`
- [ X ] The examples guidelines are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples)

This PR adds an example for using next.js with axiom.

### CODE REVIEW

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Let's convert to TypeScript. See: https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples

File: examples/with-axiom/package.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start"
+  },
+  "dependencies": {
+    "next": "latest",
+    "next-axiom": "^0.10.0",
+    "react": "^18.0.0",
+    "react-dom": "^18.0.0",
```
Comment (balazsorban44) [inline examples/with-axiom/package.json:12]:
```suggestion
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
```

File: examples/with-axiom/package.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    "swr": "^1.3.0"
+  },
+  "devDependencies": {
+    "@types/react": "^18.0.5",
+    "@types/react-dom": "^18.0.1",
+    "@types/node": "^16.11.26",
+    "typescript": "^4.6.3"
```
Comment (balazsorban44) [inline examples/with-axiom/package.json:19]:
```suggestion
    "typescript": "^4.7.4"
```

File: examples/with-axiom/pages/_app.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { log } from 'next-axiom'
+import { AppProps } from 'next/app'
+export { reportWebVitals } from 'next-axiom'
+
+log.info('Hello from frontend', { foo: 'bar' })
+
+type props = {}
```
Comment (balazsorban44) [inline examples/with-axiom/pages/_app.tsx:7]:
```suggestion
```

This seems to be unused

File: examples/with-axiom/middleware.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { NextResponse, NextRequest } from 'next/server'
+import { log, withAxiom } from 'next-axiom'
+
+async function middleware(_req: NextRequest) {
```
Comment (balazsorban44) [inline examples/with-axiom/middleware.ts:4]:
```suggestion
import { NextResponse } from 'next/server'
import { log, withAxiom } from 'next-axiom'

async function middleware() {
```

File: examples/with-axiom/.gitignore [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+yarn-error.log*
+
+# local env files
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+# vercel
+.vercel
```
Comment (balazsorban44) [inline examples/with-axiom/.gitignore:34]:
```suggestion
.vercel

# typescript
*.tsbuildinfo
```

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Let's also add a `tsconfig.json` (running `next dev` will generate one)

File: examples/with-axiom/README.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+## Deploy your own
+
+Deploy the example using [Vercel](https://vercel.com?utm_source=github&utm_medium=readme&utm_campaign=next-example) or preview live with [StackBlitz](https://stackblitz.com/github/vercel/next.js/tree/canary/examples/with-axiom)
+
+[![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/git/external?repository-url=https://github.com/vercel/next.js/tree/canary/examples/with-axiom&project-name=with-axiom&repository-name=with-axiom)
+
+Check out our [Next.js deployment documentation](https://nextjs.org/docs/deployment) for more details.
+
+## How to use
```
Comment (balazsorban44) [inline examples/with-axiom/README.md:13]: There should probably be only one `How to use` section

**Comment by schehata** [comment]
> Let's also add a `tsconfig.json` (running `next dev` will generate one)

added. (thanks for the tip, saved my time :D)

**Comment by schehata** [comment]
@balazsorban44 is there something that needs to be updated?

**Review by steven-tey — DISMISSED** [review]
LGTM! Thanks for the PR! :)

**Review by balazsorban44 — DISMISSED** [review]

**Review by balazsorban44 — APPROVED** [review]

**Pull request merged**